python/ray/ml/preprocessors/scaler.py

```diff
@@ -93,3 +94,116 @@ def column_min_max_scaler(s: pd.Series):
 
     def __repr__(self):
         return f"<Scaler columns={self.columns} stats={self.stats_}>"
+
+
+class MaxAbsScaler(Preprocessor):
+    """Scale values within columns based on the absolute max value.
+
+    For each column, each value will be transformed to ``value / abs_max``,
+    where ``abs_max`` is calculated from the fitted dataset.
+
+    When transforming the fitted dataset, transformed values will be in the
+    range [-1, 1].
+
+    Args:
+        columns: The columns that will individually be scaled.
+    """
+
+    def __init__(self, columns: List[str]):
+        super().__init__()
+        self.columns = columns
+
+    def _fit(self, dataset: Dataset) -> Preprocessor:
+        aggregates = [AbsMax(col) for col in self.columns]
+        self.stats_ = dataset.aggregate(*aggregates)
+        return self
+
+    def _transform_pandas(self, df: pd.DataFrame):
+        def column_abs_max_scaler(s: pd.Series):
+            s_abs_max = self.stats_[f"abs_max({s.name})"]
+
+            # Handle division by zero.
+            # All values are 0.
+            if s_abs_max == 0:
+                s_abs_max = 1
+
+            return s / s_abs_max
+
+        df.loc[:, self.columns] = df.loc[:, self.columns].transform(
+            column_abs_max_scaler
+        )
+        return df
+
+    def __repr__(self):
+        return f"<Scaler columns={self.columns} stats={self.stats_}>"
+
+
+class RobustScaler(Preprocessor):
+    """Scale values within columns based on their quantile range.
+
+    For each column, each value will be transformed to
+    ``(value - median) / (high_quartile - low_quartile)``,
+    where ``median`` , ``high_quartile``, and ``low_quartile``
+    are  calculated from the fitted dataset.
+
+    Args:
+        columns: The columns that will individually be scaled.
+    """
+
+    def __init__(self, columns: List[str]):
+        super().__init__()
+        self.columns = columns
+
+    def _fit(self, dataset: Dataset) -> Preprocessor:
+        # TODO(matt): Support user-defined ranges.
+        low = 0.25
+        med = 0.50
+        high = 0.75
+
+        num_records = dataset.count()
+        max_index = num_records - 1
+        split_indices = [int(percentile * max_index) for percentile in (low, med, high)]
+
+        self.stats_ = {}
+
+        # TODO(matt): Handle case where quartile lands between 2 numbers.
+        # The current implementation will simply choose the closest index.
+        # This will affect the results of small datasets more than large datasets.
```

Comment:
Not sure I follow this... Wouldn't that just output 1 boolean per quantile indicating if its an exact match?

Maybe something like this would work?
`between_indices = [(np.floor(percentile * max_index), np.ceil(percentile * max_index)) for for percentile in (low, med, high)]`